flink-core/src/main/java/org/apache/flink/core/fs/FileSystem.java

```diff
@@ -684,7 +686,9 @@ public boolean exists(final Path f) throws IOException {
 
     /**
      * Make the given file and all non-existent parents into directories. Has the semantics of Unix
-     * 'mkdir -p'. Existence of the directory hierarchy is not an error.
+     * 'mkdir -p'. Existence of the directory hierarchy is not an error. {@code FileSystem}
+     * implementations being backed by an object store like S3 do not support directories. No
+     * entities are going to be created in that case.
```

Comment:
This is only true for presto; the hadoop s3 filesystem creates marker files.

Reply:
You're right. I reverted the JavaDoc changes entirely because I feel that it adds more confusion than it helps. FLINK-26061 is still valid, though.